py/server/deephaven/table_listener.py

```diff
@@ -330,13 +330,14 @@ def __init__(self, t: Table, listener: Union[Callable, TableListener], descripti
             listener (Union[Callable, TableListener]): listener for table changes
             description (str, optional): description for the UpdatePerformanceTracker to append to the listener's entry
                 description, default is None
-            dependencies (Union[Table, Sequence[Table]]): tables that must be satisfied before the listener's execution. A
-                refreshing table is considered to be satisfied if all updates to the table have been processed in the current
-                update graph cycle. A static table is always considered to be satisfied. If a specified table is refreshing,
-                it must belong to the same update graph as the table being listened to. Default is None.
+            dependencies (Union[Table, Sequence[Table]]): tables that must be satisfied before the listener's execution.
+                A refreshing table is considered to be satisfied if all possible updates to the table have been processed
+                in the current update graph cycle. A static table is always considered to be satisfied. If a specified
+                table is refreshing, it must belong to the same update graph as the table being listened to. Default is
+                None.
 
-                Dependencies are used to ensure that the listener can safely access them during its execution, such as reading
-                the data from the tables or even performing table operations on them.
+                Dependencies are used to ensure that the listener can safely access them during its execution, such as
+                reading the data from the tables or even performing certain table operations on them.
```

Comment:
I don't think `certain` is an out here. I really don't want users thinking they should apply operations in a listener. We need to specify something like:
"While performing operations on the dependent tables is safe in the listener, reading or operating on the results of those operations may not be safe (due to memoization and other mechanisms that could cause those operations to return pre-existing results). It's best to directly depend on any table you want to consume data from or operate on, rather than assume (possibly-incorrectly) that satisfaction is transitive."

Reply:
I like the concept of the addition, but as written, this is going to go over the heads of most python users.  Can we rework this to something they can parse?

Reply:
1. If the results are static (e.g. from a `.snapshot()`).
2. If the user can guarantee that the result did not exist before this cycle. We don't give them any tools to do that besides turning off all memoization. They can use liveness scopes to throw away old results, but if the operation is applied externally it will still be memoized.